### ISSUE TITLE
Cannot allocate memory in /var/www/friendica/boot.php on line 1372

### ISSUE THREAD

**Issue by Rudloff** [friendica/friendica#3351]
Hello,

The cron sometimes returns this warning:
```
PHP Warning:  proc_open(): fork failed - Cannot allocate memory in /var/www/friendica/boot.php on line 1372
PHP Warning:  proc_close() expects parameter 1 to be resource, boolean given in /var/www/friendica/boot.php on line 1372
```
I guess it happens when I have too many things runing and not enough RAM is available on my server.

It's not a big deal but maybe we could either:
* Catch the error and retry later.
* Catch the error and log a more user-friendly message.

**Comment by annando** [comment]
This "retry later" is no problem. The worker background processing is very robust in this case.

But you are right that we should catch this error.

**Comment by Hypolite** [comment]
I've got those errors and I subsequently reduced the amount of concurrent workers from 10 to 6 on my system. I don't have those messages anymore.

**Comment by Rudloff** [comment]
I only have 4 workers and the error doesn't happen very often (just once every few days)
I'm not really worried about the error itself, it's just that I get the cron error by email and it is annoying.

**Comment by Hypolite** [comment]
> it's just that I get the cron error by email and it is annoying.

I feel you. My cron errors just are piling up in `/var/mail/` 😬

**Comment by AlfredSK** [comment]
To stop the mails from cron you can add `MAILTO=""` to your crontab.

**Comment by Hypolite** [comment]
I'm pretty sure it's not a good idea since I wouldn't have anything to debug cron executions.

**Comment by Rudloff** [comment]
Yes, I want to be notified about other errors. Just not these warnings.

**Comment by Hypolite** [comment]
#3352 removes the second warning, however the first one remains. We would need an app-wide error handler to catch it.
